include/zephyr/toolchain/armclang.h

```diff
@@ -7,6 +7,9 @@
 #ifndef ZEPHYR_INCLUDE_TOOLCHAIN_ARMCLANG_H_
 #define ZEPHYR_INCLUDE_TOOLCHAIN_ARMCLANG_H_
 
+#ifndef ZEPHYR_INCLUDE_TOOLCHAIN_H_
+#error Please include <zephyr/toolchain.h>
```

Comment:
Maybe: `Do not include toolchain headers directly, use <zephyr/toolchain.h> instead`

As it is it may sound that one should *also* include `zephyr/toolchain.h` rather than *only* including it